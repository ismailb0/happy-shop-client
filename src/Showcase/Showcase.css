.showcase {
    display: block;
    overflow-y: auto;
    position: relative;
    width: 100%;
}
