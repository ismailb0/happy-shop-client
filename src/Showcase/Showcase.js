import React, { Component } from 'react';

import ShowcaseNavigationBar from '../ShowcaseNavigationBar';
import ProductShowcase from '../ProductShowcase';
import './Showcase.css';

class Showcase extends Component {

  render() {
    return (
      <div className="showcase">
        <ShowcaseNavigationBar
          handlePageChange={this.props.handlePageChange}
          handleResultsPerPageChange={this.props.handleResultsPerPageChange}
          handleSortChange={this.props.handleSortChange}
          page={this.props.page}
          resultsPerPage={this.props.resultsPerPage}
          sortedBy={this.props.sortedBy}
          totalNumberOfProducts={this.props.totalNumberOfProducts}
        />
        <ProductShowcase
          products={this.props.products}
        />
      </div>
    );
  }
}

export default Showcase;
