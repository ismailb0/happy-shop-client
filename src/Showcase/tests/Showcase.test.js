import React from 'react';
import { shallow } from "enzyme";
import Showcase from '../Showcase';

describe('<Showcase />', () => {

  const props = {
    handlePageChange: jest.fn(),
    handleResultsPerPageChange: jest.fn(),
    handleSortChange: jest.fn(),
    sortedBy: 'price',
    resultsPerPage: 15,
    totalNumberOfProducts: 400,
    page: 1,
    products: [
      {
        brand: "Sephora Collection",
        description: "A gel lip gloss with plumped-up, light-reflecting shine.\r\n",
        image: "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
        name: "Ultra Shine Lip Gel",
        price: 19.0
      },
      {
        brand: "My Awesome Brand",
        description: "Awesome Description",
        image: "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
        name: "Awesome Name",
        price: 91.0
      },
    ]
  }

  it('should match snapshot', () => {
    const renderedComponent = shallow(<Showcase {...props} />);
    expect(renderedComponent).toMatchSnapshot();
  });

});
