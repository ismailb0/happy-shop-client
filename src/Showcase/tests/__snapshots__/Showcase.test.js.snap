// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Showcase /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Showcase
    handlePageChange={[MockFunction]}
    handleResultsPerPageChange={[MockFunction]}
    handleSortChange={[MockFunction]}
    page={1}
    products={
      Array [
        Object {
          "brand": "Sephora Collection",
          "description": "A gel lip gloss with plumped-up, light-reflecting shine.
",
          "image": "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
          "name": "Ultra Shine Lip Gel",
          "price": 19,
        },
        Object {
          "brand": "My Awesome Brand",
          "description": "Awesome Description",
          "image": "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
          "name": "Awesome Name",
          "price": 91,
        },
      ]
    }
    resultsPerPage={15}
    sortedBy="price"
    totalNumberOfProducts={400}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ShowcaseNavigationBar
          handlePageChange={[MockFunction]}
          handleResultsPerPageChange={[MockFunction]}
          handleSortChange={[MockFunction]}
          page={1}
          resultsPerPage={15}
          sortedBy="price"
          totalNumberOfProducts={400}
        />,
        <ProductShowcase
          products={
            Array [
              Object {
                "brand": "Sephora Collection",
                "description": "A gel lip gloss with plumped-up, light-reflecting shine.
",
                "image": "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
                "name": "Ultra Shine Lip Gel",
                "price": 19,
              },
              Object {
                "brand": "My Awesome Brand",
                "description": "Awesome Description",
                "image": "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
                "name": "Awesome Name",
                "price": 91,
              },
            ]
          }
        />,
      ],
      "className": "showcase",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handlePageChange": [MockFunction],
          "handleResultsPerPageChange": [MockFunction],
          "handleSortChange": [MockFunction],
          "page": 1,
          "resultsPerPage": 15,
          "sortedBy": "price",
          "totalNumberOfProducts": 400,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "products": Array [
            Object {
              "brand": "Sephora Collection",
              "description": "A gel lip gloss with plumped-up, light-reflecting shine.
",
              "image": "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
              "name": "Ultra Shine Lip Gel",
              "price": 19,
            },
            Object {
              "brand": "My Awesome Brand",
              "description": "Awesome Description",
              "image": "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
              "name": "Awesome Name",
              "price": 91,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ShowcaseNavigationBar
            handlePageChange={[MockFunction]}
            handleResultsPerPageChange={[MockFunction]}
            handleSortChange={[MockFunction]}
            page={1}
            resultsPerPage={15}
            sortedBy="price"
            totalNumberOfProducts={400}
          />,
          <ProductShowcase
            products={
              Array [
                Object {
                  "brand": "Sephora Collection",
                  "description": "A gel lip gloss with plumped-up, light-reflecting shine.
",
                  "image": "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
                  "name": "Ultra Shine Lip Gel",
                  "price": 19,
                },
                Object {
                  "brand": "My Awesome Brand",
                  "description": "Awesome Description",
                  "image": "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
                  "name": "Awesome Name",
                  "price": 91,
                },
              ]
            }
          />,
        ],
        "className": "showcase",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handlePageChange": [MockFunction],
            "handleResultsPerPageChange": [MockFunction],
            "handleSortChange": [MockFunction],
            "page": 1,
            "resultsPerPage": 15,
            "sortedBy": "price",
            "totalNumberOfProducts": 400,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "products": Array [
              Object {
                "brand": "Sephora Collection",
                "description": "A gel lip gloss with plumped-up, light-reflecting shine.
",
                "image": "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
                "name": "Ultra Shine Lip Gel",
                "price": 19,
              },
              Object {
                "brand": "My Awesome Brand",
                "description": "Awesome Description",
                "image": "https://static-reg.lximg.com/images/product_images/closeup_34208_SephoraCollection_UltraShineLipGel_01PerfectNude_WEB_a0ea7cca39eec787b338105af4ff4a85097902bf_1523780022.png",
                "name": "Awesome Name",
                "price": 91,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
